### PR TITLE
Input validation

### PR DESCRIPTION
**Pull request by SteamRiC**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,10 +98,25 @@ end
...
-    .merge(HeadersExtractor[{'authorization' => :auth}])
+    .merge(HeadersExtractor[
+      'accept' => :accept,
+      'content-type' => :content_type,
+      'x-my-value' => :my_value
+    ])
     .coerce(age: ->(age) { age.to_s })
 
   response = extractor.call(request)
+    .map(Salestation::Web::InputValidator[
```
Comment (indrekj) [inline README.md]: move to a variable like extractor

File: lib/salestation/web/input_validator.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Salestation
+  class Web < Module
+    class InputValidator
+      include Deterministic::Prelude
+
+      def self.[](**validations)
+
```
Comment (sviik) [inline lib/salestation/web/input_validator.rb]: newline

**Comment by SteamRiC** [comment]
Updated

File: lib/salestation/web/input_validator.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Salestation
+  class Web < Module
+    class InputValidator
+      include Deterministic::Prelude
+
+      def self.[](**validations)
```
Comment (indrekj) [inline lib/salestation/web/input_validator.rb]: one * is for args, two is for keyword args. should be one here

Comment (SteamRiC) [reply lib/salestation/web/input_validator.rb]: but these are keyword args, you call it like
```
Salestation::Web::InputValidator[
    accept: Salestation::Web::InputValidators::AcceptHeader['application/json', 'application/xml'],
    content_type: Salestation::Web::InputValidators::ContentTypeHeader['application/json'],
    my_value: lambda do |my_value|
      if my_value == 'foo'
        Deterministic::Result::Success(nil)
      else
        Deterministic::Result::Failure('invalid value')
      end
    end
  ]
```

Comment (indrekj) [reply lib/salestation/web/input_validator.rb]: ah my bad

**Comment by SteamRiC** [comment]
!merge

**Pull request merged**